clients/src/main/java/org/apache/kafka/server/policy/AlterConfigPolicy.java

```diff
@@ -32,6 +32,7 @@
  * using the default constructor and will then pass the broker configs to its <code>configure()</code> method. During
  * broker shutdown, the <code>close()</code> method will be invoked so that resources can be released (if necessary).
  */
+@SuppressWarnings("try") // can't override close() to now declare throws Exception until Kafka 3.0
```

Comment:
Why would we want to override `close()` to throw `Exception` if that's what it does already?

Reply:
Sorry the comment isn't too helpful. The problem is that `AlterConfigPolicy` inherits `close() from `AutoCloseable` which declares it to throw `Exception`, which includes `InterruptedException`, which generates the warning. 

/home/tom/messaging/kafka/clients/src/main/java/org/apache/kafka/server/policy/AlterConfigPolicy.java:35: warning: [try] auto-closeable resource AlterConfigPolicy has a member method close() that could throw InterruptedException

So one solution would be to enumerate more specific checked exceptions (e.g. `IOException` plus maybe others), but that would be an API change so is not an option right now. 

The alternative would be to just add the above `SuppressWarnings` and never change the `throws` clause. The drawback of that would be that it doesn't address the warning that's generated at the use site.

See also: https://bugs.openjdk.java.net/browse/JDK-8155591

What's your preference?